verilog/analysis/checkers/enum_name_style_rule.cc

```diff
@@ -40,39 +44,54 @@ using verible::LintViolation;
 using verible::SyntaxTreeContext;
 using verible::matcher::Matcher;
 
-static constexpr absl::string_view kMessage =
-    "Enum names must use lower_snake_case naming convention "
-    "and end with _t or _e.";
+static constexpr absl::string_view style_default_regex = "[a-z_0-9]+(_t|_e)";
 
-const LintRuleDescriptor& EnumNameStyleRule::GetDescriptor() {
+EnumNameStyleRule::EnumNameStyleRule()
+    : style_regex_(
+          std::make_unique<re2::RE2>(style_default_regex, re2::RE2::Quiet)),
+      violation_message_(absl::StrCat(
+          "Enum type name does not match the naming convention ",
+          "defined by regex pattern: ", style_regex_->pattern())) {}
+
+const LintRuleDescriptor &EnumNameStyleRule::GetDescriptor() {
   static const LintRuleDescriptor d{
       .name = "enum-name-style",
       .topic = "enumerations",
       .desc =
-          "Checks that `enum` names use lower_snake_case naming convention "
-          "and end with '_t' or '_e'.",
+          "Checks that enum type names follow a naming convention defined by "
+          "a RE2 regular expression.\n"
+          "Example common regex patterns:\n"
```

Comment:
Instead of repeating the same examples in various places using regexps, how about we add a little section in the lint [README.md](https://github.com/chipsalliance/verible/tree/master/verilog/tools/lint#readme) with regexp examples and just add a link here ?

(also here, the default regex, the one with the trailing _e or _t is not even mentioned)

Reply:
No problem.